lib/openapi_parser/schema_validators/base.rb

```diff
@@ -17,5 +17,23 @@ def initialize(validatable, coerce_value)
     def coerce_and_validate(_value, _schema)
       raise 'need implement'
     end
+
+    def validate_discriminator_schema(discriminator, value)
+      unless value.key?(discriminator.property_name)
+        return [nil, OpenAPIParser::NotExistDiscriminatorPropertyName.new(discriminator.property_name, value, discriminator.object_reference)]
+      end
+      mapping_key = value[discriminator.property_name]
+
+      # TODO: it's allowed to have discriminator without mapping, then we need to lookup discriminator.property_name
+      # but the format is not the full path, just model name in the components
+      mapping_target = discriminator.mapping[mapping_key]
+      unless mapping_target
+        return [nil, OpenAPIParser::NotExistDiscriminatorMappingTarget.new(mapping_key, discriminator.object_reference)]
+      end
+
+      # TODO: this is likely O(n) for each validation, we need to find it once during parsing
+      resolved_schema = discriminator.root.find_object(mapping_target)
```

Comment:
@ota42y so I am trying to move this to https://github.com/Ladas/openapi_parser/blob/c844e37dba4cbe605f9a3f8d565f84843acfe5bb/lib/openapi_parser/schemas/schema.rb#L115, to my own parser, but after spending several hours there I didn't figure out how it works. :-) Could you give me some hints? Is it even possible to do the lookup there?

If the `discriminator.root.find_object(mapping_target)` is or would be O(1), I don't really need to do that. :-)

Reply:
it looks like the find_object does O(n) lookup at worst, then caches the result, so this should be ok performance wise